lstchain/calib/camera/drs4.py

```diff
@@ -1,68 +1,71 @@
+                                 meanped, numped):
         size4drs = 4096
         roisize = 40
-        for i in range(0, 2):
-            for j in range(0, 7):
-                fc = int(first_cap[i, j])
-                pixel =  expected_pixel_id[nr*7 + j]
-                posads0 = int((2+fc)%size4drs)
-                if posads0 + 40 < 4096:
-                    meanped[i, j, posads0:(posads0+36)] += waveform[i, pixel, 2:38]
-                    numped[i, j, posads0:(posads0 + 36)] += 1
-                else:
-                    for k in range(2, roisize-2):
-                        posads = int((k+fc)%size4drs)
-                        val = waveform[i, pixel, k]
-                        meanped[i, j, posads] += val
-                        numped[i, j, posads] += 1
+        for nr_module in prange(0, 265):
+            first_cap = first_cap_array[nr_module, :, :]
+            for gain in prange(0, 2):
+                for pix in prange(0, 7):
+                    fc = first_cap[gain, pix]
+                    pixel = expected_pixel_id[nr_module * 7 + pix]
+
+                    posads0 = int((2 + fc) % size4drs)
+                    if posads0 + 40 < 4096:
+                        meanped[gain, pixel, posads0:(posads0 + 36)] += waveform[gain, pixel, 2:38]
+                        numped[gain, pixel, posads0:(posads0 + 36)] += 1
+
+                    else:
+                        for k in prange(2, roisize - 2):
+                            posads = int((k + fc) % size4drs)
+                            val = waveform[gain, pixel, k]
+                            meanped[gain, pixel, posads] += val
+                            numped[gain, pixel, posads] += 1
 
     def finalize_pedestal(self):
-        try:
-            self.meanped = self.meanped/self.numped
-        except Exception as err:
-            print("Not enough events to coverage all capacitor. Please use more events to create pedestal file.")
-            print(err)
+        if np.sum(self.numped==0) > 0:
+            raise RuntimeError("Not enough events to coverage all capacitor. "
+                               "Please use more events to create pedestal file.")
+        else:
+            self.meanped = self.meanped / self.numped
 
     def get_first_capacitor(self, event, nr):
-        fc = np.zeros((2, 8))
-        first_cap = event.lst.tel[self.telid].evt.first_capacitor_id[nr * 8:(nr + 1) * 8]
+        fc = np.zeros((2, 7))
+        first_cap = event.lst.tel[self.tel_id].evt.first_capacitor_id[nr * 8:(nr + 1) * 8]
         # First capacitor order according Dragon v5 board data format
         for i, j in zip([0, 1, 2, 3, 4, 5, 6], [0, 0, 1, 1, 2, 2, 3]):
-            fc[self.high_gain, i] = first_cap[j]
+            fc[0, i] = first_cap[j] # high gain
         for i, j in zip([0, 1, 2, 3, 4, 5, 6], [4, 4, 5, 5, 6, 6, 7]):
-            fc[self.low_gain, i] = first_cap[j]
+            fc[1, i] = first_cap[j] # low gain
```

Comment:
why changing the clear general definition of high/low gain by these two numbers (and corresponding comments)?